devel/libsignal-protocol-c/Portfile

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        signalapp libsignal-protocol-c 2.3.3 v
+revision            0
+checksums           rmd160  26c279a15203f11f92edab1ceb64de1c62339dd5 \
+                    sha256  eb8ef165a22582c00f9c31fdbd435dd787451c495d772302d08fd9afc0de96fb \
+                    size    272197
+
+categories          devel security net
+platforms           darwin
+license             {GPL-3 MPL-2.0}
+maintainers         {isi.edu:calvin @cardi} openmaintainer
+description         A ratcheting forward secrecy protocol for \
+                    synchronous and asynchronous messaging environments.
+long_description    ${description}
+
+variant tests description {Enable unit tests} {
+    depends_lib-append      port:check \
+                            port:openssl
```

Comment:
Is this incompatible with libressl? If not, or if you don't know, assume it is compatible and use a `path:` dependency here. See other ports that depend on openssl.

Are these really library dependencies or are they only needed to run the tests? If the latter, use `depends_test`.

Reply:
I'll test with libressl. I went with openssl based on the [upstream build dependencies](https://github.com/signalapp/libsignal-protocol-c#build-dependencies).

The dependencies are just for tests, but I'm not sure if they're required in the build phase as `-DBUILD_TESTING=1` is passed to the top-level `cmake` for the tests variant. I'll check and test this.

Reply:
> I'll test with libressl. I went with openssl based on the [upstream build dependencies](https://github.com/signalapp/libsignal-protocol-c#build-dependencies).

Right, and it doesn't say there that it doesn't work with libressl, so since we offer multiple openssl implementations, make sure to use a path-style dependency so that either of them could satisfy it.

But their instructions do also say: "On MacOS X, Common Crypto is used instead of OpenSSL"

> The dependencies are just for tests, but I'm not sure if they're required in the build phase as `-DBUILD_TESTING=1` is passed to the top-level `cmake` for the tests variant. I'll check and test this.

That may be a good point.

Reply:
> But their instructions do also say: "On MacOS X, Common Crypto is used instead of OpenSSL"

Based on their instructions, the default install does not require libressl/openssl–it's only required for their unit tests. I've confirmed that the `tests` variant compiles and runs using the `path:` dependency to libressl.

> Are these really library dependencies or are they only needed to run the tests? If the latter, use depends_test.

I put the dependencies under `depends_test` and confirmed that `port test` works with my setup (I uninstalled `port:check` to verify).